src/metrics.ts

```diff
@@ -68,6 +72,8 @@ export class Metrics {
     this.#minHostRate = minHostRate;
     this.#maxHostRate = maxHostRate;
     this.#minHostRequests = Math.round(minHostRate * (maxAge/1000));
+    this.#maxHostRatio = Math.max(Math.min(maxHostRatio, 0.9), 0); // 90% is very high, but this cap is just to prevent invalid ratios
+    this.#hostRatioMaxCount = Math.ceil(maxHostRatio * 100 * 10); // 10x requests compared to host ratio (10% * 10 = 100)
```

Comment:
Should this be calculated based on `this.#maxHostRatio` instead of the raw `maxHostRatio` (after the min/max has been applied)?

Reply:
good catch